src/component/archive/ArchiveTable/ArchiveTable.tsx

```diff
@@ -0,0 +1,360 @@
+                    style={{
+                        height:
+                            rowHeight * rows.length +
+                            theme.shape.tableRowHeightCompact,
+                    }}
+                >
+                    <SortableTableHeader
+                        headerGroups={headerGroups as any}
+                        flex
+                    />
+                    <TableBody {...getTableBodyProps()}>
+                        {rows.map((row, index) => {
+                            const isVirtual =
+                                index < firstRenderedIndex ||
+                                index > lastRenderedIndex;
+
+                            if (isVirtual) {
+                                return null;
+                            }
+
+                            prepareRow(row);
+                            return (
+                                <TableRow
+                                    hover
+                                    {...row.getRowProps()}
+                                    style={{
+                                        display: 'flex',
+                                        top:
+                                            index * rowHeight +
+                                            theme.shape.tableRowHeightCompact,
+                                    }}
+                                    className={classes.row}
+                                >
+                                    {row.cells.map(cell => (
+                                        <TableCell
+                                            {...cell.getCellProps({
+                                                style: {
+                                                    flex: cell.column.minWidth
+                                                        ? '1 0 auto'
+                                                        : undefined,
+                                                },
+                                            })}
+                                            className={classes.cell}
+                                        >
+                                            {cell.render('Cell')}
+                                        </TableCell>
+                                    ))}
+                                </TableRow>
+                            );
+                        })}
+                    </TableBody>
+                </Table>
+            </SearchHighlightProvider>
+            <ConditionallyRender
+                condition={rows.length === 0 && searchValue?.length > 0}
+                show={
+                    <TablePlaceholder>
+                        No feature toggles found matching &ldquo;
+                        {searchValue}&rdquo;
+                    </TablePlaceholder>
+                }
+            />
+            <ConditionallyRender
+                condition={rows.length === 0 && searchValue?.length === 0}
```

Comment:
Repetitive `rows.length` check. Could have a wrapper `ConditionallyRender` that would check `rows.length` and then a `ConditionallyRender` inside checking for `searchValue?.length` with a `show` and `elseShow` props. Check the described pattern in `FeatureToggleListTable`.